app/coffee/Features/Announcements/AnnouncementsHandler.coffee

```diff
@@ -5,20 +5,44 @@ _ = require("lodash")
 logger = require("logger-sharelatex")
 settings = require("settings-sharelatex")
 
-module.exports =
 
-	getUnreadAnnouncements : (user_id, callback = (err, announcements)->)->
+
+
+module.exports = AnnouncementsHandler = 
+
+
+
+	_domainSpecificAnnouncements : (email)->
+		domainSpecific = _.filter settings?.domainAnnouncmentsToShow, (domainAnnouncment)->
+			matches = _.filter domainAnnouncment.domains, (domain)->
+				return email.indexOf(domain) != -1
+			return matches.length > 0
+		return domainSpecific or []
+
+
+	getUnreadAnnouncements : (user, callback = (err, announcements)->)->
 		async.parallel {
 			lastEvent: (cb)->
-				AnalyticsManager.getLastOccurance user_id, "announcement-alert-dismissed", cb
+				AnalyticsManager.getLastOccurance user?._id, "announcement-alert-dismissed", cb
```

Comment:
At this point, if the user doesn't exist we're presumably not going to anything sensible from this? So perhaps just a top level null user guard that bails out with an error, rather than potentially failing silently?